File: level2/문자열압축.cpp
#include <string>
#include <vector>

using namespace std;

int getDigit(int n){
    int digit = 0;
    while(n){
        digit++;
        n /= 10;
    }
    return digit;
}

int solution(string s) {
    int s_len = s.size();
    int answer = s_len;
    for(int l = 1; l <= s_len/2; l++){
        int idx = 0, cnt = s_len;
        while(idx < s_len){
            string str = s.substr(idx,l);
            int tmp_idx = idx+l, s_cnt = 1;
            for(int i = tmp_idx; i < s_len; i+=l){
                if(str != s.substr(i,l)){
                    break;
                }
                s_cnt++;
            }
            idx += s_cnt * l;
            cnt -= (s_cnt-1)*l;
            if(s_cnt != 1){
                cnt += getDigit(s_cnt);
            }
        }
        if(answer > cnt){
            answer = cnt;
        }
    }
    return answer;
}

File: level2/문자열압축.py
def solution(s):
    answer = 1001
    
    iter_len = 1
    s_len = len(s)
    while iter_len <= int(s_len/2) + 1:
        compact_s = ''
        pre_s = s[:iter_len]
        
        num = 1
        i = iter_len
        while i < s_len:
            cur_s = s[i:i+iter_len]
            i += iter_len
            
            if cur_s != pre_s:
                if num == 1:
                    compact_s += pre_s
                else:
                    compact_s += str(num)+pre_s
                    
                num = 1
                pre_s = cur_s
                continue
                
            num += 1
            
        if num == 1:
            compact_s += pre_s
        else:
            compact_s += str(num)+pre_s
            
        iter_len += 1
        
        answer = min(answer,len(compact_s))
    
    return answer
